cyberdrop_dl/scraper/crawlers/tiktok_crawler.py

```diff
@@ -0,0 +1,119 @@
+from __future__ import annotations
+
+from typing import TYPE_CHECKING
+
+from aiolimiter import AsyncLimiter
+from yarl import URL
+
+from cyberdrop_dl.scraper.crawler import Crawler, create_task_id
+from cyberdrop_dl.utils.data_enums_classes.url_objects import FILE_HOST_PROFILE, ScrapeItem
+from cyberdrop_dl.utils.utilities import error_handling_wrapper, get_filename_and_ext
+
+if TYPE_CHECKING:
+    from collections.abc import AsyncGenerator
+
+    from cyberdrop_dl.managers.manager import Manager
+
+
+class TikTokCrawler(Crawler):
+    primary_base_domain = URL("https://tiktok.com/")
+
+    def __init__(self, manager: Manager) -> None:
+        super().__init__(manager, "tiktok", "TikTok")
+        self.api_url = URL("https://www.tikwm.com/api/")
+        self.request_limiter = AsyncLimiter(1, 10)
+
+    @create_task_id
+    async def fetch(self, scrape_item: ScrapeItem) -> None:
+        if any(part in scrape_item.url.parts for part in ("video", "photo")):
+            await self.video(scrape_item)
+        elif "@" in scrape_item.url.parts[1]:
+            await self.profile(scrape_item)
+
+    async def profile_post_pager(self, scrape_item: ScrapeItem) -> AsyncGenerator[dict]:
+        username = scrape_item.url.parts[1][1:]
+        cursor = "0"
+        while True:
+            posts_api_url = (self.api_url / "user" / "posts").with_query(
+                {"cursor": cursor, "unique_id": username, "count": "50"}
+            )
+            async with self.request_limiter:
+                json_data = await self.client.get_json(self.primary_base_domain, posts_api_url, origin=scrape_item)
+
+            if scrape_item.album_id is None:
+                author_id = json_data["data"]["videos"][0]["author"]["id"]
+                scrape_item.album_id = author_id
+                scrape_item.add_to_parent_title(self.create_title(username))
+
+            yield json_data
+
+            if not json_data["data"]["hasMore"]:
+                break
+            cursor = json_data["data"]["cursor"]
+
+    @error_handling_wrapper
+    async def handle_image_post(self, scrape_item: ScrapeItem, post: dict) -> None:
+        post_id = post["video_id"]
+        title = post["title"] if post["title"] else f"Post {post_id}"
+        new_scrape_item = self.create_scrape_item(
+            scrape_item, scrape_item.url, title, True, post_id, post["create_time"]
+        )
+        for image_url in map(URL, post["images"]):
+            filename, ext = get_filename_and_ext(image_url.name)
+            scrape_item.add_children()
+            await self.handle_file(image_url, new_scrape_item, filename, ext)
```

Comment:
Only increase children count after they are handled
```suggestion
            await self.handle_file(image_url, new_scrape_item, filename, ext)
            scrape_item.add_children()
```